ravenframework/Optimizers/RavenSampled.py

```diff
@@ -231,7 +231,7 @@ def localGenerateInput(self, model, inp):
       self.inputInfo['batchMode'] = False
     for _ in range(self.batch):
       inputInfo = {'SampledVarsPb':{}, 'batchMode':self.inputInfo['batchMode']}  # ,'prefix': str(self.batchId)+'_'+str(i)
-      if self.counter == self.limit + 1:
+      if self.getIteration(all(self._activeTraj)) == self.limit + 1:
```

Comment:
I'm not familiar with what `all` does here. Doesn't that do a binary check on each entry of `self._activeTraj`? Did you want:
```python
if all(self.getIteration(self._activeTraj) == self.limit + 1)
```
perhaps?